src/main.test.ts

```diff
@@ -0,0 +1,437 @@
+            path.join(project.directoryPath, 'src'),
+          );
+        }
+        const stdout = new MockWritable();
+        const stderr = new MockWritable();
+
+        await main({
+          argv: ['node', 'module-lint', ...projectNames],
+          stdout,
+          stderr,
+          config: {
+            cachedRepositoriesDirectoryPath,
+            defaultProjectNames: [],
+          },
+        });
+
+        const output = stdout.data().map(stripAnsi).join('');
+
+        expect(output).toBe(
+          `
+repo-1
+------
+
+Linted project in 0 ms.
+
+- Does the \`src/\` directory exist? ✅
+
+repo-2
+------
+
+Linted project in 0 ms.
+
+- Does the \`src/\` directory exist? ✅
+
+`,
+        );
+      });
+    });
+
+    it('lists the rules executed against the default repositories which fail', async () => {
+      await withinSandbox(async ({ directoryPath: sandboxDirectoryPath }) => {
+        const projectNames = ['repo-1', 'repo-2'];
+        const { cachedRepositoriesDirectoryPath } =
+          await setupToolWithMockRepositories({
+            execaMock,
+            sandboxDirectoryPath,
+            repositories: [
+              { name: 'metamask-module-template', create: true },
+              ...projectNames.map((projectName) => ({
+                name: projectName,
+                create: true,
+              })),
+            ],
+          });
+        const stdout = new MockWritable();
+        const stderr = new MockWritable();
+
+        await main({
+          argv: ['node', 'module-lint', ...projectNames],
+          stdout,
+          stderr,
+          config: {
+            cachedRepositoriesDirectoryPath,
+            defaultProjectNames: [],
+          },
+        });
+
+        const output = stdout.data().map(stripAnsi).join('');
+
+        expect(output).toBe(
+          `
+repo-1
+------
+
+Linted project in 0 ms.
+
+- Does the \`src/\` directory exist? ❌
+  - \`src\` exists in the module template, but not in this repo.
+
+repo-2
+------
+
+Linted project in 0 ms.
+
+- Does the \`src/\` directory exist? ❌
+  - \`src\` exists in the module template, but not in this repo.
+
+`,
+        );
+      });
+    });
+
+    it('does not exit immediately if a project fails to lint for any reason, but shows the reason and continues', async () => {
+      await withinSandbox(async ({ directoryPath: sandboxDirectoryPath }) => {
+        const projectNames = ['repo-1', 'repo-2'];
+        const { cachedRepositoriesDirectoryPath } =
+          await setupToolWithMockRepositories({
+            execaMock,
+            sandboxDirectoryPath,
+            repositories: [
+              { name: 'metamask-module-template', create: true },
+              ...projectNames.map((projectName) => ({
+                name: projectName,
+                create: false,
+              })),
+            ],
+            validRepositories: [
+              {
+                name: 'repo-1',
```

Comment:
Nit: Perhaps we should reverse the order here, and make repo-2 fail? Otherwise this test might pass even if the tool exited when encountering the failure (since the failure is on the last repo in the list)

Reply:
Okay, changed here: [dda0390](https://github.com/MetaMask/module-lint/pull/33/commits/dda0390bad4dda628c0e5c8a9a317c24c27b0d9f)